### PR TITLE
Implement Lesson API Client

### PR DESCRIPTION
**Pull request by y2kwak**
## Implementation Description
Adding a lesson api client! 

## Screenshots
N/A

## What should reviewers focus on?
- ~~Do we need the `bearerToken` like in [EntityAPIClient](https://github.com/uwblueprint/rowan-house/blob/main/frontend/src/APIClients/EntityAPIClient.ts#L35-L42)?~~
- Ensure that request and response types are appropriate.

## Testing Procedure
- What test(s) should we run to make sure your code works?
   - What ways could the input be wrong?
   - Is it secure? Could nefarious users access/break it?

- Need to implement unit tests on all the methods

## Things to Note
_- Need to implement better error handling_

## Checklist
- [x] Ensure code follows style guide by running the linters
- [x] I have updated the documentation or deemed it unnecessary
- [x] I have linked the relevant issue in this PR
- [x] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import baseAPIClient from "./BaseAPIClient";
+
+export type LessonRequest = {
+  course: string;
+  title: string;
+  description: string;
+  image: string;
+  content: [Record<string, unknown>];
```
Comment (Shrinjay) [inline frontend/src/APIClients/lessonAPIClient.ts:10]: Wouldn't `any` be more appropriate than `unknown` here?

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+// TO DO: error handling
+
+const get = async (id: string): Promise<LessonResponse> => {
+  try {
```
Comment (Shrinjay) [inline frontend/src/APIClients/lessonAPIClient.ts:29]: I'm pretty sure we need to include the bearer token here like we do with EntityAPIClient, that token will be needed for any authenticated routes which the lessons route should be?

**Review by Shrinjay — CHANGES_REQUESTED** [review]
Bearer token probably does need to be included, otherwise just 1 minor thing

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import baseAPIClient from "./BaseAPIClient";
```
Comment (e-wai) [inline frontend/src/APIClients/lessonAPIClient.ts:1]: Can we rename file to `LessonAPIClient.ts` (caps different)?

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+export type LessonRequest = {
+  course: string;
+  title: string;
+  description: string;
+  image: string;
+  content: [Record<string, any>];
+};
+
+export type LessonResponse = {
+  id: string | number;
```
Comment (e-wai) [inline frontend/src/APIClients/lessonAPIClient.ts:16]: won't it always be a string?

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  course: string;
+  title: string;
+  description: string;
+  image: string;
+  content: [Record<string, any>];
+};
+
+// TO DO: error handling
+
+const getLesson = async (id: string): Promise<LessonResponse> => {
```
Comment (e-wai) [inline frontend/src/APIClients/lessonAPIClient.ts:26]: Do we want any other getters? Off the top of my head,
- `get` -> returns all lessons
- `getLessonsByCourse` -> returns all lessons belonging to a course

File: frontend/src/APIClients/lessonAPIClient.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  const bearerToken = `Bearer ${getLocalStorageObjProperty(
+    AUTHENTICATED_USER_KEY,
+    "accessToken",
+  )}`;
+  try {
+    const { data } = await baseAPIClient.get(`/lesson${id}`, {
```
Comment (e-wai) [inline frontend/src/APIClients/lessonAPIClient.ts:32]: What's the difference between `/lesson/${id}` and `/lesson${id}`? Starter code seems to follow the former

**Review by e-wai — CHANGES_REQUESTED** [review]
Looks pretty good -- couple of questions, but nothing major

File: frontend/src/APIClients/lessonAPIClient.ts (hunk 6 above)
Comment (y2kwak) [reply frontend/src/APIClients/lessonAPIClient.ts:32]: Good catch! It should be `/lesson/${id}`

File: frontend/src/APIClients/lessonAPIClient.ts (hunk 5 above)
Comment (y2kwak) [reply frontend/src/APIClients/lessonAPIClient.ts:26]: Currently, we only have the [getLessonById](https://github.com/uwblueprint/rowan-house/blob/main/backend/services/implementations/lessonService.ts#L15) in the back end. It's outlined in[ the ticket ](https://github.com/uwblueprint/rowan-house/issues/55)that retrieving all lessons of a module should be implemented separately
But if it makes sense, maybe this getter should be renamed to `getLessonById`.. lmk what you think!

File: frontend/src/APIClients/lessonAPIClient.ts (hunk 4 above)
Comment (y2kwak) [reply frontend/src/APIClients/lessonAPIClient.ts:16]: I noticed that the [EntityResponse](https://github.com/uwblueprint/rowan-house/blob/main/frontend/src/APIClients/EntityAPIClient.ts#L21) type implemented it as string or number. 

But yeah, I would assume it'll always be a string so I'll remove the or number

**Review by draymond63 — APPROVED** [review]
Looks good to me! I'll let others review, but if they're happy with it, it can be merged

File: frontend/src/APIClients/lessonAPIClient.ts (hunk 5 above)
Comment (e-wai) [reply frontend/src/APIClients/lessonAPIClient.ts]: Meh it's pretty minor to me. Whatever is consistent/clear with other usages in repo

**Review by e-wai — APPROVED** [review]
Seems good to me -- how'd you test this?

**Review by Shrinjay — APPROVED** [review]

**Pull request merged**